axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceTermServiceImpl.java

```diff
@@ -903,6 +904,51 @@ protected InvoiceTerm createNewTerm(
     return invoiceTermRepo.save(newInvoiceTerm);
   }
 
+  @Transactional
+  protected InvoiceTerm createNewTerm(InvoiceTerm originalInvoiceTerm, BigDecimal amount) {
+    InvoiceTerm newInvoiceTerm = new InvoiceTerm();
+    newInvoiceTerm.setIsCustomized(true);
+    newInvoiceTerm.setIsPaid(false);
+    newInvoiceTerm.setMoveLine(originalInvoiceTerm.getMoveLine());
+    newInvoiceTerm.setDueDate(originalInvoiceTerm.getDueDate());
+    newInvoiceTerm.setIsHoldBack(originalInvoiceTerm.getIsHoldBack());
+    newInvoiceTerm.setEstimatedPaymentDate(originalInvoiceTerm.getEstimatedPaymentDate());
+    newInvoiceTerm.setAmount(amount);
+    newInvoiceTerm.setAmountRemaining(amount);
+    newInvoiceTerm.setPaymentMode(originalInvoiceTerm.getPaymentMode());
+    newInvoiceTerm.setBankDetails(originalInvoiceTerm.getBankDetails());
+    newInvoiceTerm.setPfpValidateStatusSelect(InvoiceTermRepository.PFP_STATUS_AWAITING);
+    newInvoiceTerm.setPfpValidatorUser(originalInvoiceTerm.getPfpValidatorUser());
+    newInvoiceTerm.setPfpGrantedAmount(BigDecimal.ZERO);
+    newInvoiceTerm.setPfpRejectedAmount(BigDecimal.ZERO);
+    return invoiceTermRepo.save(newInvoiceTerm);
+  }
+
+  public void createInvoiceTerm(
+      MoveLine creditMoveLine,
+      BankDetails bankDetails,
+      User pfpUser,
+      PaymentMode paymentMode,
+      LocalDate todayDate,
+      BigDecimal amountRemaining) {
+    InvoiceTerm newInvoiceTerm = new InvoiceTerm();
+    newInvoiceTerm.setIsCustomized(true);
+    newInvoiceTerm.setIsPaid(false);
+    newInvoiceTerm.setDueDate(todayDate);
+    newInvoiceTerm.setIsHoldBack(false);
+    newInvoiceTerm.setEstimatedPaymentDate(null);
+    newInvoiceTerm.setAmount(amountRemaining);
+    newInvoiceTerm.setAmountRemaining(amountRemaining);
+    newInvoiceTerm.setPaymentMode(paymentMode);
+    newInvoiceTerm.setBankDetails(bankDetails);
+    newInvoiceTerm.setPfpValidateStatusSelect(InvoiceTermRepository.PFP_STATUS_AWAITING);
+    newInvoiceTerm.setPfpValidatorUser(pfpUser);
+    newInvoiceTerm.setPfpGrantedAmount(BigDecimal.ZERO);
+    newInvoiceTerm.setPfpRejectedAmount(BigDecimal.ZERO);
+    newInvoiceTerm.setPercentage(BigDecimal.valueOf(100));
+    creditMoveLine.addInvoiceTermListItem(newInvoiceTerm);
```

Comment:
please move the duplicated code into a third method to create invoice term, `new InvoiceTerm()` must appear only once